### PR TITLE
Fix selections with inherited axes

### PR DESCRIPTION
**Pull request by jrs65**
This fixes the issues which was with the incorrect `._axes` attribute being checked because the correct `.axes` method can't be called from a class method. This sorts all of that out.

### CODE REVIEW

**Comment by jrs65** [comment]
Requires radiocosmology/caput#123

File: test/test_selections.py [hunk 1]
```diff
@@ -87,37 +88,44 @@ def container_on_disk_distributed():
...
-    assert np.all(m["gain"][:] == dset1[(fsel, isel, slice(None))])
-    assert np.all(m["weight"][:] == dset2[(fsel, slice(None))])
+    m = GainData.from_file(
+        container_on_disk, freq_sel=fsel, input_sel=isel, time_sel=tsel
+    )
+    assert np.all(m["gain"][:] == dset1[(fsel, isel, tsel)])
+    assert np.all(m["weight"][:] == dset2[(fsel, tsel)])
     assert np.all(m.index_map["freq"] == freqs[fsel])
     assert np.all(m.index_map["input"] == inputs[isel])
+    assert np.all(m.index_map["time"] == inputs[tsel])
```
Comment (ssiegelx) [inline test/test_selections.py:98]:
```suggestion
    assert np.all(m.index_map["time"] == times[tsel])
```

File: test/test_selections.py [hunk 2]
```diff
@@ -87,37 +88,44 @@ def container_on_disk_distributed():
...
+        time_sel=tsel,
+        distributed=True,
     )
-    assert np.all(m["gain"][:] == dset1[(fsel, isel, slice(None))])
-    assert np.all(m["weight"][:] == dset2[(fsel, slice(None))])
+    assert np.all(m["gain"][:] == dset1[(fsel, isel, tsel)])
+    assert np.all(m["weight"][:] == dset2[(fsel, tsel)])
     assert np.all(m.index_map["freq"] == freqs[fsel])
     assert np.all(m.index_map["input"] == inputs[isel])
+    assert np.all(m.index_map["time"] == inputs[tsel])
```
Comment (ssiegelx) [inline test/test_selections.py:115]:
```suggestion
    assert np.all(m.index_map["time"] == times[tsel])
```

File: test/test_selections.py [hunk 3]
```diff
@@ -87,37 +88,44 @@ def container_on_disk_distributed():
...
-        )
+        md = GainData.from_file(container_on_disk_distributed, distributed=False)
 
-        assert np.all(md["gain"][:] == dset1[(fsel, isel, slice(None))])
-        assert np.all(md["weight"][:] == dset2[(fsel, slice(None))])
+        assert np.all(md["gain"][:] == dset1[(fsel, isel, tsel)])
+        assert np.all(md["weight"][:] == dset2[(fsel, tsel)])
         assert np.all(md.index_map["freq"] == freqs[fsel])
         assert np.all(md.index_map["input"] == inputs[isel])
+        assert np.all(md.index_map["time"] == inputs[tsel])
```
Comment (ssiegelx) [inline test/test_selections.py:128]:
```suggestion
        assert np.all(md.index_map["time"] == times[tsel])
```

File: test/test_selections.py (hunk 1 above)
Comment (jrs65) [reply test/test_selections.py:98]: Ooops. This is a fail! I'll post a new commit to fix all this up.

**Pull request merged**